Remove password_hash from returned user object in user controllers.

JadziaHax opened the pull request:
Both user objects returned by getUserById and getUserByEmail no longer
have the password_hash as a property.+

JadziaHax committed:
Remove password_hash from returnted user object in user controllers.

Both user objects returned by getUserById and getUserByEmail no longer
have the password_hash as a property.+

perfectlynormalbeast commented:
I would consider using the `delete` operator (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete) to delete the password_hash key instead of rewriting the entire object out?

Also, in the interest of DRY, maybe write that as a one-line function `cleanUserObject()`?

JadziaHax committed:
Move user object scrubbing to a helper function.

Note, the helper function as written violates this eslint rule:
http://eslint.org/docs/rules/no-param-reassign

however if you read it we can pass an option to it in the eslint config
to alow property modifications.

JadziaHax commented:
@perfectlynormalbeast I've moved it into a helper function that uses the delete operator, but this shouldn't pass linting because the air-bnb config disallows functions that modify or reassign parameters per this rule: http://eslint.org/docs/rules/no-param-reassign. But we can configure eslint with an option for this rule that will allow property modification.

perfectlynormalbeast commented:
Does the linter still complain if you do something like
```
let cleanUser = user;
delete cleanUser.password_hash;
return cleanUser;
```
?

JadziaHax commented:
Ah no, if we remove the single line constraint this becomes much easier.

JadziaHax committed:
Make the cleanUserObject use a copy of the user parameter.

JadziaHax committed:
Merge branch 'master' of https://github.com/Twilio-org/phonebank into return-user-no-password
